transport/conn.go

```diff
@@ -383,6 +383,10 @@ func OpenShardConn(ctx context.Context, addr string, si ShardInfo, cfg ConnConfi
 	it := ShardPortIterator(si)
 	maxTries := (maxPort-minPort+1)/int(si.NrShards) + 1
 	for i := 0; i < maxTries; i++ {
+		if err := ctx.Err(); err != nil {
+			return nil, fmt.Errorf("aborted opening connection on shard %d: %w", si.Shard, err)
+		}
+
```

Comment:
Does the second commit (`transport: conn, abort OpenShardConn faster when context is done`) belong to this PR? It looks like it can be submitted separately and reviewed independently.

Reply:
Moved to #286.